abcclassroom/github.py

```diff
@@ -77,9 +78,12 @@ def check_git_ssh():
             stderr=subprocess.PIPE,
         )
     except subprocess.CalledProcessError as e:
-        # We ALWAYS will get here, because that subprocess call returns
+        # We ALWAYS will get here. If you include check=True in subprocess
+        # it returns a calledprocess error. We do this because otherwise
+        # it's harder to capture a failed call. It just fails quietly. With
+        # check=True subprocess call returns
         # a non-zero exit code whether ssh access is set up correctly or
-        # not. Must check output.
+        # not. We then parse output to see what was returned.
```

Comment:
hey @kcranston i need a bit of advice here. i'm having a hard time triggering a filenotfounderror for testing. it seems to always go to that final `else` statement when i trigger any type of error. Just wondering how you managed to get to that part of this conditional? any advice is welcome. i will comment out the test that i have for now and move on but am hopeful you have some ideas!

Reply:
Oh gosh. It's possible that there is a bug here. I recall going through a lot of contortions to get my machine _not_ set up correctly in order to trigger the various failures. Testing this requires uninstallation of open-ssh.

Reply:
ahhhh ok i think i understand. I *think* i also found a simpler version of the mock test code that is reproducible that i can share. i will try to pull that together so you can kind of just run it to see what works and doesn't work. i've managed to test the entire function with the exception of that last part. we have to mock it because so many users will already have ssh setup.  (i think) but it should work properly on CI because ssh won't be setup there i think